connector/extensions/web-service/src/main/java/org/oaebudt/edc/web/service/MetadataValidator.java

```diff
@@ -0,0 +1,68 @@
+package org.oaebudt.edc.web.service;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class MetadataValidator {
+
+    private static final Pattern EMAIL_REGEX = Pattern.compile(
+            "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}$", Pattern.CASE_INSENSITIVE
+    );
+
+    public static ValidationResult validate(Map<String, Object> metadata) {
+        ValidationResult result = ValidationResult.ok();
+
+        result.merge(validateRequiredString(metadata, "legalOrganizationName"));
+        result.merge(validateRequiredString(metadata, "countryOfOrganization"));
+        result.merge(validateRequiredString(metadata, "contactPerson"));
+        result.merge(validateRequiredString(metadata, "dataProcessingDescription"));
+        result.merge(validateRequiredString(metadata, "qualityAssuranceMeasures"));
+        result.merge(validateRequiredString(metadata, "dataLicensingTerms"));
+
+        result.merge(validateUrl(metadata, "organizationWebsite"));
+        result.merge(validateEmail(metadata, "contactEmail"));
+
+        result.merge(validateLevel(metadata, "dataAccuracyLevel"));
+        result.merge(validateLevel(metadata, "dataGenerationTransparencyLevel"));
+        result.merge(validateLevel(metadata, "dataDeliveryReliabilityLevel"));
+        result.merge(validateLevel(metadata, "dataFrequencyLevel"));
+        result.merge(validateLevel(metadata, "dataGranularityLevel"));
+        result.merge(validateLevel(metadata, "dataConsistencyLevel"));
+
+        return result;
+    }
+
+
+    private static ValidationResult validateRequiredString(Map<String, Object> map, String key) {
```

Comment:
another refactoring suggestion for the future would be to model the validation with the composition pattern, that will make this code more modular.
As an example you can take a look at the [`JsonObjectValidator`](https://github.com/eclipse-edc/Connector/blob/f491d2fe83a69e113bdcb2f89b8c0f556e555ef9/core/common/lib/validator-lib/src/main/java/org/eclipse/edc/validator/jsonobject/JsonObjectValidator.java), it's an higher level of complexity, but the concept is: the validator is built once through the builder setting the rules, then it's called passing only the object under validation (in our case, the `metadata` map)

Reply:
Not sure I understand @ndr-brt 
The JsonObjectValidator has a builder. 
Not sure I totally understand the concept